lib/facter/util/linux/dhcp.rb

```diff
@@ -75,6 +75,11 @@ def search_internal_leases(interface_name)
           def search_with_dhcpcd_command(interface_name)
             @log.debug("Attempt to get DHCP for interface #{interface_name}, from dhcpcd command")
 
+            if @dhcpcd_command_exists.nil?
+              @dhcpcd_command_exists = !Facter::Core::Execution.execute('which dhcpcd', logger: @log).empty?
+            end
+            return unless @dhcpcd_command_exists
+
             output = Facter::Core::Execution.execute("dhcpcd -U #{interface_name}", logger: @log)
```

Comment:
I know it's out of scope for this ticket (dhcpcd is not functional/present), but maybe we can optimize calls to dhcpcd as well? (In a different ticket)

```
     -U, --dumplease [interface]
             Dumps the current lease for the interface to stdout.  If no interface is given then all interfaces are dumped.  Use the -4 or -6 flags to specify an address family.  If a lease is piped in via standard input
             then that is dumped.  In this case, specifying an address family is mandatory.
```

Reply:
It might not really work as documented:
```
user@machine:~ $ sudo dhcpcd -U eth0
broadcast_address='192.168.200.255'
dhcp_lease_time='259200'
dhcp_message_type='5'
dhcp_server_identifier='192.168.200.1'
domain_name_servers='192.168.200.1'
ip_address='192.168.200.3'
network_number='192.168.200.0'
routers='192.168.200.1'
subnet_cidr='24'
subnet_mask='255.255.255.0'
dhcp6_client_id='0001000125d83160dca33344bac4'
dhcp6_name_servers='fe80::1%eth0'
dhcp6_server_id='00010001dc43e795443335916115'
user@machine:~ $ sudo dhcpcd -U 
: dhcp_dump: Invalid argument

```